docs/UserGuide.md

```diff
@@ -1,225 +1,342 @@
+### View
+For: Show information about the module with the specified `MODULE_CODE`. In doing so, the app will automatically
+ switch to the specific module's tab.
 
-Examples: `T info n/CS2100 midterm`
+Format: `M view m/MODULE_CODE`
 
-### Add
-Description: Adds a task to TrackIt@NUS
+Examples: `M view m/CS2103T`
 
-Format: `T add n/{task name} d/{task date} a/{location} w/{weightage} r/{remark}`
+Remark: The `MODULE_CODE` provided must be present in the Module list.
 
-Examples: `T add n/Assignment 1 due d/20/11/2020 a/NUS w/20 r/Focus on Chapters 1-3`
+### Add
 
-### Edit
-Description: Edits the detail of a task.
+For: Add a new module.
 
-Format: `T edit {event_index} n/{event name} d/{event date}`
+Format: `M add m/MODULE_CODE n/NAME`
 
-Examples: `T edit 1 n/Assignment 1 due t/22/11/2020`
+Examples: `M add m/CS1231S n/Discrete Structures`
 
-### Delete
-Description: Deletes a task from the list.
+### Edit
+For: Edit the module with the specified `MODULE_CODE`.
 
-Format: `T delete {event_index}`
+Format: `M edit m/MODULE_CODE [m/NEW_MODULE_CODE] [n/NAME]`
 
-Examples: `T delete 2`
+Examples: `M edit m/CS1231S m/CS1101S n/Programming Methodology`
 
-### List
-Description: Lists all events.
+Remark: The `MODULE_CODE` provided must be present in the Module list. At least 1
+ field must be provided for editing. i.e. `M edit m/CS2100` is not allowed.
 
-Format: `T list`
+### Delete
+For: Delete a module with the specified `MODULE_CODE`.
 
-Examples: `T list`
+Format: `M delete m/MODULE_CODE`
 
-## Lesson
+Examples: `M delete m/CS2100`
 
-### View
-For: Shows the information about a lesson
+Remark: The `MODULE_CODE` provided must be present in the Module list.
 
-Format: `L info m/{module_code} t/{type}`
-
-Examples:
-* `L info m/CS2103T t/tutorial`
-* `L info m/CS2103T t/lesson`
+## Lesson <a name="lesson"></a>
 
 ### Add
-For: Add a lesson to a module. The module needs to be existing prior to adding.
+For: Add a lesson to a module.
 
-Format: `L add m/{module_code} n/{type} t/{time} v/{venue or Zoom link}`
+Format: `L add m/MODULE_CODE n/TYPE d/DATE a/LOCATION`
 
-`{time}` : the time of the first lesson of that type in this semester
+Examples: `L add m/CS2103T n/tutorial d/2pm - 3pm Wed a/COM1`
 
-Examples: `L add m/CS2103T n/tutorial t/01/08/2019 4pm-6pm v/COM1`
+Remark: The module must exist prior to adding a lesson. `DATE` must be of the form `START_TIME - END_TIME DAY_OF_WEEK`.`TYPE` must be one of
+ the following:
+* `lecture`
+* `tutorial`
+* `lab`
+* `sectional`
+* `recitation`
 
 ### Edit
 
 For: Edit the detail of a lesson
 
-Format: `L edit m/{module_code} [n/{type}] [t/{time}] [v/{venue or Zoom link}]`
+Format: `L edit INDEX [m/MODULE_CODE] [n/TYPE] [d/DATE] [a/LOCATION]`
+
+Examples: `L edit 1 m/CS2103T n/tutorial d/2pm - 3pm Monday`
 
-Examples: `L edit m/CS2103T n/tutorial t/02/08/2019 4pm-6pm v/COM1`
+Remark: The `INDEX` provided must be positive and cannot be larger than the length of the Lesson list. At least 1
```

Comment:
"the length of the Lesson list" sounds a little unnatural, maybe "the number of lessons"?